### ISSUE TITLE
Installing Win98

### ISSUE THREAD

**Issue by grstatdoc-gr** [joncampbell123/dosbox-x#4328]
### Question

How do I install win98 when I have the iso file in the DOSBox-X directory?
I have created a hdd image file, but am having problems after that:
IMGMOUNT D Win98.iso doesn't work (I get the "The image must be on a host, local or network drive. Must specify image file(s) to mount" error
So, I'm stumped as to access the iso file to continue the process

### Have you checked that no similar question(s) exist?

- [X] I have searched and didn't find any similar question.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by rderooy** [comment]
As mentioned on the Win98 installation guide on the wiki, you specify the host path to the ISO file. If the file (on the host) is in C:\ISO you would do an `imgmount d C:\ISO\Win98.iso` to mount it.

**Comment by grstatdoc-gr** [comment]
Some progress...mounted the HDD drive and the ISO on D...
but I've run into the SU0409 error, and the Windows 98 SE: Setup Detection
Notice Message SU0409 #123 thread doesn't seem to help me get past that.
Mike Dean

**Comment by rderooy** [comment]
Are you following the dosbox-x wiki guide on installing Win98?

**Comment by grstatdoc-gr** [comment]
Doing my best....
Mike Dean

**Comment by grstatdoc-gr** [comment]
I am "boxed" into a corner with the Win98 installation...
Even though I downloaded a bootable Win98 iso, I've run into the "El Torito CD-ROM boot record not found"
So, I've tried Method 2, and have run into the SU0409 error
While I really don't want to have to turn off my antimalware to finish the install, I may have to do so...

**Comment by grstatdoc-gr** [comment]
I did a win98 iso that does have a boot image (overcame that hurdle), but here is the next one:
<img width="685" alt="image" src="https://github.com/joncampbell123/dosbox-x/assets/71800985/a6529e5f-9ac0-41a8-baaa-3fd16853237f">

**Comment by grstatdoc-gr** [comment]
These are the steps I've taken:
<img width="425" alt="image" src="https://github.com/joncampbell123/dosbox-x/assets/71800985/9f7e123c-5476-4437-b894-7eff8af99016">
This is before I typed the BOOT A: command

**Comment by grstatdoc-gr** [comment]
Okay, I've reached this point:
Now start DOSBox-X as follows to continue the installation process:

dosbox-x -conf win98.conf

The wiki isn't clear to me (it may be to everybody else) if I'm to  type setup here:
<img width="433" alt="image" src="https://github.com/joncampbell123/dosbox-x/assets/71800985/37e5f007-9dfb-455a-8380-02b372b88ebd">

Another question about regedit:
_Once the files are copied, start REGEDIT and navigate to HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Setup and change SourcePath= to the location where you copied the files. e.g., SourcePath=C:\WIN98_
Is it safe to add the SourcePath key as it doesn't exit in my Registry.

I haven't found this in the wiki (not sure where this is), but how do I get back to Z: from the guest machine in DOSBox-X?

These are the steps I've distilled for myself:
1. Open Command Prompt
2. Run dosbox-x -conf win98.conf
3. From DOSBox-X window: 
IMGMAKE hdd.img -t hd -size 16384 (I think I can skip this step)
run IMGMOUNT C hdd.img  (I think I can skip this step, too)
IMGMOUNT D C:\dosroot\Win98.iso
IMGMOUNT A -bootcd D
BOOT A:

I don't usually ask for gentleness here, but since I'm fighting mental blocks in following the steps, some is welcome.

Wondering if there is a way of pasting commands in the DOSBox-X window...

**Comment by grstatdoc-gr** [comment]
However, if I get the message that the drive is already mounted, when I
rerun these steps...I don't have to recreate the drive and remount a drive
that is already mounted, do I?????
Mike Dean

**Comment by rderooy** [comment]
You should not do the `mount c c:\dosroot` first. Or before you issue the `boot` command, you must it with remove it with something like `mount c -u` and then `imgmount` the harddisk image file.

**Comment by grstatdoc-gr** [comment]
dosroot is where I will do most of my operations (since DOSBOX-X does not recommend mounting the entire C drive, and the programs I intend on running are stored there)
But finally, I got Win98 to install. Thanks for your help.
I guess the last thing I need to ask is how to start DOSBox-X in Win98 without having to open the command prompt to do it every time.

**Comment by grstatdoc-gr** [comment]
<img width="676" alt="image" src="https://github.com/joncampbell123/dosbox-x/assets/71800985/5d6eb86b-f523-42c3-82c1-3041bd9a993e">
Not OK for me...
Mental block:
I guess I need to copy the win98 folder to the hdd.img, but am not sure how to do it with xcopy.

**Comment by grstatdoc-gr** [comment]
<img width="475" alt="image" src="https://github.com/joncampbell123/dosbox-x/assets/71800985/2b8dd5c8-deb9-400b-b8d1-15f380850036">

Progress, I think,,,,
Still having problems getting the boot folder onto the hdd.img

**Comment by rderooy** [comment]
No. you are still doing the `mount` command, which causes the following `imgmount` command to fail.

**Comment by grstatdoc-gr** [comment]
I follow (now that I had commented out that mount command). Thanks.
It took a while for me to understand that Z: in DOSBox-X is the C:\DOSBox-X folder (as I was reading the steps, I had the belief that I had to put the needed files somewhere else to get it to work)

**Comment by grstatdoc-gr** [comment]
Finally -- thanks, Robert, for putting up with my stumbles and missteps -- I am running Windows 98 in DOSBox-X!
But now, I could use a road map to:
1. Install the .net versions suggested.
2. Resize the Win98 screen after starting it so that it wouldn't distort the text
3. Copy file into the Win98 C:\ folder, as I'm not seeing how to do that once I have Win98 running.

thanks!

**Comment by grstatdoc-gr** [comment]
Not sure if I ought to request that this question be closed and start a new one in Windows ME....
But, in a few words, I have installed successfully WinMe in my Windows 11 DOSBox-X, added the network configuration info to winme.conf. 
But from the WinMe end, I am not sure if I can get it to see the NE200 as a network card, or if I'm trying to do something that is not possible (my reading the networking section in the wiki has, unfortunately, left my head spinning, and WinMe is expecting a NIC card, which isn't feasible(?) in a guest system)
I think this is the last step I need to solve to get TAS Baseball to run optimally (I can run the program in WinMe and Win98 and print reports, but is the best way to get the reports off the guest system to use the copy command from the command prompt? )
Thanks.

**Comment by rderooy** [comment]
It is possible to get NE2000 networking working in WinME. But the emulated NE2000 card will not get autodetected, as it does not implement PnP extensions. As such you must manually add it as documented on the wiki.
https://dosbox-x.com/wiki/Guide%3AInstalling-Windows-ME#_enabling_networking

But for this to work, the NE2000 emulation must be enabled in your config **AND** working on your host, which depends on the type of backend you chose (PCAP or SLIRP). In particular PCAP, while more capable, can be difficult or even impossible to setup in some environments.

Once again, this is a pretty advanced topic.

Also as an FYI, using WinME will not make any difference compared to Win98 as to the supported SMB version. It will still be limited to SMBv1.

**Comment by grstatdoc-gr** [comment]
Windows ME does not seem to like the IRQ = 10 and I/O range = 0300-031F
<img width="401" alt="image" src="https://github.com/joncampbell123/dosbox-x/assets/71800985/c54b1837-2556-4401-b78c-e357abbf8346">
I am using SLIRP on the backend

**Comment by grstatdoc-gr** [comment]
Somewhat of an annoyance, following the first error:
<img width="400" alt="image" src="https://github.com/joncampbell123/dosbox-x/assets/71800985/8e4b34f8-96e8-4595-a1c5-a55a8f2ad37b">

**Comment by grstatdoc-gr** [comment]
Well, it was a try on Windows 98...
Mike Dean

**Comment by grstatdoc-gr** [comment]
I had supposed that the ne2000.sys driver was already in the Win98SE iso,
but it looks like I will have to add it, since my Win98 can't find it.
Mike Dean

**Comment by grstatdoc-gr** [comment]
Solved the driver problem, sort of...
Win98 reports that NE 2000 Compatible "is either not present, not working
properly, or does not have all the drivers installed (Code 10)"
The current driver provider is Microsoft, and the date is 4-23-1999, which
is the best driver for the system.
The resources indicate IRQ of 10 and I/O range of 0300-031F.

Is there a setting in the config file that also addresses the Standard
IDE/ESDI Hard Disk Controller or, since it is a guest machine off a virtual
hard drive, can be ignored?
Thanks.

Mike Dean

**Comment by rderooy** [comment]
The NE2000 driver **is** included with Win95, 98 and ME. There are no third party network drivers to install. The issue could be either with the host configuration, dosbox-x config file or win9x configuration.

**Comment by grstatdoc-gr** [comment]
Ok....
Checking the configuration for the NE 2000 card in Win 98:
IRQ = 10; I/O range = 0300-031F

In the win98.conf file:
ne2000 = true; nicirq = 10; nicbase = 300

(I figure the dosbox-x.conf file is not used when launching DOSBOX-X in
win98 mode.
I'm hoping I don't have to delete everything and start from scratch, seeing
that I installed win98 first and later added the ne2000 info ...
Mike Dean

**Comment by grstatdoc-gr** [comment]
I'm not sure if a properly configured NE2000 adapter is not supposed to
show up in Device Manager, but after setting it up with the IRQ of 10 and
I/O range of 300-31F, it doesn't show....
I would expect the driver to be in the Win98/drivers folder, but I haven't
found it...

Under Hard Disk Controllers, I have one with:
IRQ=15, I/O range=0170-0177 and 0376-0377
and another one that was installed (when I got the NE200 going) with:
IRQ=14; I/O range=01F0-01F7 and 03F6-03F7
It's this latter one that "is either not present, not working properly or
does not have all the drivers installed (Code 10)"
Mike Dean

**Comment by grstatdoc-gr** [comment]
So, I can safely rule out the lack of a current driver and IRQ-I/O
settings, leaving me somewhat stymied
Mike Dean

**Comment by rderooy** [comment]
This is using the SLIRP backend on DOSBox-X SDL2 2023.05.01 on Linux.

![image](https://github.com/joncampbell123/dosbox-x/assets/17701098/6ea55adc-5142-42d5-8ec1-d1f40fb20373)

NE2000 Settings in the associated dosbox-x config are:
```ini
[ne2000]
ne2000=true
nicirq=10
macaddr=AC:DE:48:88:99:AD
backend=slirp

[etnernet, pcap]
enable=false

[ethernet, slirp]
enable=true
```

**Comment by rderooy** [comment]
Also, can you try to start DOSBox-X, but before you issue the `boot` command, check if `Z:\SYSTEM\NE2000.COM` exists? If it does not, than your DOSBox-X binary may have been compiled without NE2000 emulation support.

**Comment by grstatdoc-gr** [comment]
Hmmmmm...
I get a "Bad command or file name" when I look for the file
And, the only item I have in the drivez folder is README.TXT
So, if my DOSBox-X binary was compiled w/o NE2000 emulation support, how do I correct this?

In my win98.conf file:
[ne2000]
ne2000  = true
# nicbase = 300
nicirq  = 10
macaddr = AC:DE:48:88:99:AA
backend = slirp

[ethernet, pcap]
enable=false
# realnic = list
# timeout = default

[ethernet, slirp]
enable=true
# ipv4_network    = 10.0.2.0
# ipv4_netmask    = 255.255.255.0
# ipv4_host       = 10.0.2.2
# ipv4_nameserver = 10.0.2.3
# ipv4_dhcp_start = 10.0.2.15

is it safe to uncomment the lines I have commented out?
Thanks! this has been quite helpful

**Comment by grstatdoc-gr** [comment]
Looked in the wrong place for that file...
<img width="764" alt="image" src="https://github.com/joncampbell123/dosbox-x/assets/71800985/5e41cc84-f89a-48a5-b006-dab8517c9295">
but no, that file is not there, so I need to correct this.

**Comment by rderooy** [comment]
You will only see the `Z:\SYSTEM\NE2000.COM` binary if...
- the dosbox-x binary is compiled with NE2000 & SLIRP support 
- you started dosbox-x with the NE2000 emulation options enabled in the corresponding dosbox-x config file.
 
If both of these are not true, the `NE2000.COM` will not be there.

Since you are on Windows, are you using the VS (Visual Studio) or MinGW compiled binary? I don't use Windows myself, but I think NE2000 and SLIRP may only work with one of the two.

**Comment by grstatdoc-gr** [comment]
OK, the next task is finding one of those builds...or is it already included in the dosbox-x-winXP-2023.05.01-setup.exe file?

**Comment by grstatdoc-gr** [comment]
I found it under releases...so got that to work(?)

**Comment by grstatdoc-gr** [comment]
However, I have found that while I was successful in IMGMOUNT C hdd.img under the previous dosbox setup, with the newer MinGW setup, I get the "unable to create drive from file" error.
(I would have expected the imgmount command to work in either case...)

**Comment by grstatdoc-gr** [comment]
I seem to have enough space on my hard drive, but in remaking the hdd.img file, I am getting this error:

<img width="739" alt="image" src="https://github.com/joncampbell123/dosbox-x/assets/71800985/1ba81098-00c7-4e93-9f82-04c7235e6dd4">

**Comment by grstatdoc-gr** [comment]
Well, I started over from scratch and tried the MinGW option, but couldn't recreate the hdd.img file.
I'll start over again and try the VS one.

**Comment by grstatdoc-gr** [comment]
Guess I can conclude that I can created the 16GB hdd in the VS option, but not the MinGW one in Windows 11

**Comment by grstatdoc-gr** [comment]
Struggling to remember that I have to run the DOSBox-X with the Win98.conf file **before** trying to create large hdd files!

**Comment by grstatdoc-gr** [comment]
So, I got the NE2000 Compatible to install, IRQ=10, I/O range=300-31F in both the config file and the Win98 OS, backend=slirp in the config file, and still can't pinpoint why the NE2000 Compatible isn't working correctly (Code 10)

**Comment by rderooy** [comment]
Do you at least see the `Z:\SYSTEM\NE2000.COM` file before you issue the `boot` command? If you don't, it will never work.

Also, you did not need to re-create your HDD image file. All you needed to do was to run a device discovery in Win98 for it to detect the NE2000 adapter.

**Comment by grstatdoc-gr** [comment]
This is the contents of z:\system (I commented out the boot command in the win98,config file):
<img width="723" alt="image" src="https://github.com/joncampbell123/dosbox-x/assets/71800985/4cd501b0-80a7-4710-b2d1-7359c33c0de3">

**Comment by grstatdoc-gr** [comment]
I am finding that when I run the mingw_lowend_sdl1 version, I get the "cannot create drive from file" error when I run the imgmount c hdd.img command.  The ne2000.com file isn't in the z:\system directory, either

**Comment by grstatdoc-gr** [comment]
so now things reduce to "what can I do to get the ne2000.com file in the z:system directory", since uninstalling and reinstalling DOSBox-x didn't work for me?

**Comment by rderooy** [comment]
Here is what I get:
![image](https://github.com/joncampbell123/dosbox-x/assets/17701098/52150e70-9462-4e87-a837-f43fc302721a)

**Comment by grstatdoc-gr** [comment]
I'll give that a look shortly

**Comment by grstatdoc-gr** [comment]
<img width="330" alt="image" src="https://github.com/joncampbell123/dosbox-x/assets/71800985/06121efa-def7-429b-afb7-50d01e2ef5f0">

commands done....still doesn't want to show up:
<img width="721" alt="image" src="https://github.com/joncampbell123/dosbox-x/assets/71800985/3b24fb63-a66e-4d39-be50-6453d33af40d">

not sure at all how to find/generate the log output for the two get commands ... I do have a messages file generated, but the get files output wasn't in there.

**Comment by rderooy** [comment]
The `config -get ne2000` for you clearly shows `ne2000=false` meaning it being disabled. So `Z:\SYSTEM\NE2000.COM` not showing is expected.

**Comment by grstatdoc-gr** [comment]
I see that now...
My next question is overcoming the inability to mount hdd.img under MinGW sdl1 (I recall that sdl2 is not recommended....)

**Comment by grstatdoc-gr** [comment]
Ok...I misunderstood the config -get command...it returns what is in the config file.
But, I have ne2000=true in the config file
<img width="266" alt="image" src="https://github.com/joncampbell123/dosbox-x/assets/71800985/e29ca380-d595-41de-87b4-8060f4a7f8d4">
... but getting ne2000.com to show up in the z:\system directory is what I need to solve
<img width="605" alt="image" src="https://github.com/joncampbell123/dosbox-x/assets/71800985/78518435-4336-446a-be83-081bead514f2">

**Comment by grstatdoc-gr** [comment]
I put a reference to the mingw low end in my win98 batch file, but I am still stumped by cannot create drive from file error:
<img width="362" alt="image" src="https://github.com/joncampbell123/dosbox-x/assets/71800985/f3044bc8-4818-4129-a222-41c758c32c6e">
(in addition to the ne2000.com file not being in the system directory)

**Comment by rderooy** [comment]
Yes the `config -get` output is **influenced** by your config file. But clearly the `ne2000=true` setting in your config file is being ignored. And it is likely being ignored because your dosbox-x binary lacks NE2000 or SLIRP support. Which is why it is not showing 'true'. So if you solve the reason for the setting being ignored you will solve both issues.

**Comment by rderooy** [comment]
Can anyone else check with the MinGW binary if NE2000 + SLIRP works on Windows? I don't have any Windows.

**Comment by rderooy** [comment]
For the "cannot create drive from file" error. Can you run the `ver` command, and show the output?

**Comment by grstatdoc-gr** [comment]
Well, I had hoped the running the MinGW build would solve that...but I have no real idea of finding a dosbox-x binary that has both ne2000 and slirp support.
Without that, the hope of running TAS Baseball in DOSBox-x will have to be scuttled (since it can only run in a 32-bit environment)

**Comment by grstatdoc-gr** [comment]
Another trip to the config file, as ver reports?:
DOSBox-X version 2023.05.01 (SDL1). Reported DOS version 7.10

**Comment by rderooy** [comment]
It is not a trip to the config file. The config file gets read on startup only. Meaning you are showing the values in memory.

I wanted to check if it was actually showing DOS 7.1 since the filesystem on the HDD is likely FAT32 which is only supported with DOS 7.1 or later. I suspect what you ran into is a bug in the MinGW build, possibly this one: https://github.com/joncampbell123/dosbox-x/issues/4355

**Comment by grstatdoc-gr** [comment]
Ok, thanks for the clarification.
I read about that bug, and had the impression that there is a build that
fixed the problem, but I haven't latched on to it.

**Comment by grstatdoc-gr** [comment]
So, is there a build I can download that fixes bug #2355?
Also, is there a current download that includes ne2000.com, since the download I have never had that file?

**Comment by grstatdoc-gr** [comment]
I did find the ne2000.com file and added it to my z: directory (I had hoped to put it in the z:\system directory, but that will require a binary that contains that file....

**Comment by Torinde** [comment]
The 2355 is not a bug, but is about a dialog box from DOSbox-X (requesting confirmation when closing  "are you sure you want to quit?") that can be made not to appear as [described there](https://github.com/joncampbell123/dosbox-x/issues/2355#issuecomment-796895512).

**Comment by rderooy** [comment]
I installed Win10 in a virtual machine, and then the mingw sdl2 version of DOSBox-X 2023.05.01.

I used my existing win98 HDD image that I created on Linux.

I encountered the following issues.
- my HDD image file, which mounts without issues on Linux with the same DOSBox-X version, failed to mount on Windows unless I specified the size parameter (same issue with the VS build).
- Starting Win98 I was greeted with a message that there was a problem with the NE2000 adapter
- Mouse behaved very erratic, making it uncontrollable. It directly swerved to the Win98 screen edges when captured.

I then took a step back and installed the mingw sdl2 version of DOSBox-X 2023.03.31. This had the following effects.
- on startup Scandisk complained that my filesystem was corrupted, and complained that the LBA may be incorrect. I let it proceed to run a sector check and it claimed it could not reach part of the disk. I cancelled the suggested fixes and let Win98 boot.
- No NE2000 error on startup, and I was able to ping successfully to the internet from a DOS command prompt.
- Still the same erratic mouse behaviour

Regarding the NE2000 issues, it seems that the Windows MinGW build of version 2023.05.01 (as available from the installer package) was indeed build without NE2000 support.

**Comment by grstatdoc-gr** [comment]
I have found that the main version also does not have ne2000 support

**Comment by rderooy** [comment]
Not sure what you mean with "main" version. There is no version called as such. If you mean the VS build, then yes that is at least a known limitation as documented.

**Comment by grstatdoc-gr** [comment]
Yes, I  am referring to the VS one (in contrast to the others appended with
MinGW, SDL)

**Comment by rderooy** [comment]
There seems to be some confusion about SDL. To clarify, **all** dosbox versions use SDL. DOSBox-X in particular is available in either a SDL1 or SDL2 edition.

SDL2 is newer, and at least on Linux systems more commonly used. But some people still seem to prefer the SDL1 based version. This was originally because the SDL2 based version lacked some features, but I don't think that is the case any longer. And since SDL is amongst others responsible for input devices, using the newer SDL version can actually solve some issues.

**Comment by grstatdoc-gr** [comment]
I found the revised mingw low end assembly and downloaded it....
I'm still running into the cannot create drive from file error....
<img width="480" alt="image" src="https://github.com/joncampbell123/dosbox-x/assets/71800985/3093b828-f746-4bb7-a1ac-c2bae27d13a2">
Wasn't it recommended to use MinGW_lowend_SDL1 in running Windows 98? Has it been tried with MinGW_lowend_SDL2?
I'll give SDL2 a try and see what happens...

**Comment by grstatdoc-gr** [comment]
Looking at the state of things:
Wondering if there will be a new version with NE2000 support (adding ne2000.com to drive Z  didn't get it to work properly in Win98) ... or if there is a solution to the "cannot create drive from file" error under v2023.05.01, MinGW SDL1, 32-bit.

A side question: how do I get the title bar to show which type of DOSBox-X is running, whether it's the MinGW SDL1, 32-bit or one of the other types (like Windows SDL1, 32-bit)?

**Comment by rderooy** [comment]
I don't build the Windows version, so cannot answer that. But going to an older version fixed it for me. Perhaps also try the non-installer Windows builds that you can download from the github releases page.

As mentioned before, the NE2000.COM is not used by Win98 period. It is a DOS network driver, not a Windows network driver. All I wanted to know if you could see the file, as that at least indicates that NE2000 is actually enabled. Therefore adding it yourself will not change anything.

You can manually update the title using the `title=` option in the config file, but it is manual, so it will not automatically add VS or MinGW to it.

**Comment by grstatdoc-gr** [comment]
I will try the 2023.04.01 mingw low end release and see how that works.
I had hoped that adding it would would have solved the problem of the driver not working right when I boot up Win98, but no such happenstance.
I still don't know to solve the problem of ne2000 not working correctly once I boot up Win98 (the error message doesn't give sufficient clues of how to address it, and reinstalling it did not solve the problem.

**Comment by grstatdoc-gr** [comment]
I had paused pursuing this topic when baseball tournaments for me started, but am back to probing problems with the NE2000 adapter....the 2023.04.02 mingw low end release did not solve the problem of that driver not working right (I've downloaded the 2023.10.06 one to be current)...
Well, I don't know how to get the Win98/Me windows in DOSBOX-X to send files to the host, as that is needed to run The Automated ScoreBook program (it runs OK in itself) properly.
Has anyone else who has run 98/Me been able to send files to the host?
